### PR TITLE
34 local hierarchy

### LINKED ISSUE #32: Read-hierarchy-by-id API endpoint
### Needs
* #4 

### Description
Create a read-hierarchy-by-id API endpoint, to serve the local hierarchy of a node. Return all of 6 levels above and 6 levels below such that the selected node is a direct descendant / ancestor. Only data needed for graphing should be sent - URIs, labels, deprecation date, as well as what is needed to interpret the hierarchical structure. No need to deal with multiple parents for now, do as you wish (let the team know).

The intention is to hide / show the correct levels on the frontend. Hopefully local hierarchies won't be as heavy as the full graph, but let the team know if there are still concerns about size. 

Most likely this will use similar functionality to 
* #55 

can think of reusing some functions maybe?

### To-do
* [x] Get 6 levels above and below a node from our database
* [x] Convert to nested json in the format the frontend wants
* [x] Create endpoint and serve local hierarchy data

### LINKED ISSUE #34: Local hierarchy visualisation
### Needs
* #32 
* #26 

### Description
When a node in the graph view is clicked, get its local hierarchy from the backend read-many endpoint and render it in the graph view. Allow for deselection to return to the default full view. Also get the number of levels displayed to change according to the search sidepane's level filter inputs. 

### To-do
* [x] Get local hierarchy data from read-many endpoint
* [ ] Render local hierarchy in graph view on select, full RDL on deselect
* [ ] Render only hierarchy levels contained in the level filter inputs

### LINKED ISSUE #37: Deprecation indicator
### Needs
* #36 

### Description
There should be a way to be alerted that a class has been deprecated, alongside the date of deprecation. This should be obvious both in the information sidepane as well as nodes on the main graph view (?). Design TBD - as long as it looks nice and also won't be missed. This information should already be passed in from the backend.

### To-do
* [x] Deprecation indicator in information sidepane
* [x] Deprecation indicator in main graph view

### PR DESCRIPTION
**Pull request by paulmmaingi**
### Description
The local hierarchy for any searched node has been implemented. The global and local views have been merged into 1 double sided graph that draws forward for children and backwards for parents. Default view/view on first instance is the root node of the graph, but the search functionality allows access to any local node hierarchy. Levels up and down have been dropped due to the use of dynamic expansion on loading both the children and parent hierarchies.  Separate colours denote varying nodes on the graph, i.e. expandable, root, deprecated, parent, child.

### To-do's Completed

>* [x] Get local hierarchy data from read-many endpoint
>* [x] Render local hierarchy in graph view on select, full RDL on deselect - both views are now 1
>* [x] Render only hierarchy levels contained in the level filter inputs - this has been dropped and replaced by the dynamic loading
>* [x] Deprecation indicator in information sidepane
>* [x] Deprecation indicator in main graph view

### Linked Issues

Closes #37 
Closes #34 
Closes #32  - hierarchy has been dropped for double sided tree and relevant controller updates are on this branch

### CODE REVIEW

File: src/client/src/components/GraphSearchSidepane.vue [hunk 1]
```diff
@@ -214,11 +220,11 @@ export default {
 
           <div v-if="isLeftExpanded" class="m-4 mb-5">
             <label class="flex items-center text-white mb-[30px] mt-7 whitespace-nowrap">
-              <input type="checkbox" class="mr-2" v-model="showDeprecated" />
+              <input type="checkbox" class="mr-2" v-model="deprecatedToggle" />
```
Comment (keyman015) [inline src/client/src/components/GraphSearchSidepane.vue:223]: The search window is not affected by the deprecation toggle, would it be possible to update the API route to have `dep={'true' or 'false'}` using the deprecatedToggle variable?

File: src/client/src/views/GraphView.vue [hunk 2]
```diff
@@ -96,76 +94,16 @@ function toggleIsRightExpanded() {
...
-const showLabelsInGraph = ref(true)
-
-// Toggles whether labels are displayed in the graph.
-function handleToggleLabels(value: boolean) {
-  showLabelsInGraph.value = value
-}
 </script>
 
 <template>
   <div class="container">
```
Comment (keyman015) [inline src/client/src/views/GraphView.vue:100]: Can `class="container"` be removed as it causes the visual bug on the graphing view with the invisible bar blocking the graph.

File: src/client/src/components/GraphSearchSidepane.vue (hunk 1 above)
Comment (paulmmaingi) [reply src/client/src/components/GraphSearchSidepane.vue:223]: Will be fixed on a new branch so as to get local hierarchy in asap

**Review by keyman015 — APPROVED** [review]
If you're leaving the search to another PR thats fine.

File: src/client/src/assets/apiFunctions.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+ * @throws TypeError if a network error occurred
+ * @throws Error if any other error occurred (invalid response, invalid node, etc.)
+ */
+async function fetchChildren(node: any, includeDeprecated: boolean = false): Promise<any> {
+  if (!node || !node.id) {
+    console.error('Invalid node:', node)
+    return null
+  }
+  if (!node.has_children) {
+    // console.log('Node has no children:', node)
```
Comment (sellsol) [inline src/client/src/assets/apiFunctions.ts:24]: stray console.log

File: .husky/pre-commit [hunk 4]
```diff
@@ -4,11 +4,52 @@
...
+    break
+  fi
+done
+
 # Navigate to the server directory
-cd ../../src/server
+cd src/server
 
-# Activate the Python virtual environment
-. venv/bin/activate
```
Comment (sellsol) [inline .husky/pre-commit:11]: Would the previous one not have worked for Windows? Would just like to know. Just seems like dealing with compatibility would be a hassle

**Review by sellsol — CHANGES_REQUESTED** [review]

**Review by sellsol — APPROVED** [review]

**Pull request merged**